test/integration/test/webpush_test.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+const chakram = require('chakram'), expect = chakram.expect;
+const Config = require('config-js');
+
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var Prepper = require('../lib/testPrepper.js');
+
+Prepper.makeSuite('Test Push Service locally',function(){
+  var header = new Config('./test/integration/lib/config/header.js');
+  var credential = config.get('credential'); 
+
+  var setupURL = config.get('foxbox.url') + '/users/setup';
+  var loginURL = config.get('foxbox.url') + '/users/login';
+  var serviceURL = config.get('foxbox.url') + '/api/v1';
+  var serviceListURL = serviceURL + '/services'; 
+  var getterURL = serviceURL + '/channels/get';
+  var setterURL = serviceURL + '/channels/set';
+    
+  it('Create a new credential',function(){
+
+    return chakram.post(setupURL,credential)
+    .then(function(setupResp){
+      expect(setupResp).to.have.status(201);
+    });
+  });
+
+  describe ('Once logged in', function () {
+    
+    before(function() {
+      var encoded_cred = new Buffer(credential.username+
+      ':'+credential.password).toString('base64');
+      header.Authorization = 'Basic ' + encoded_cred;
+
+      return chakram.post(loginURL,null,{'headers' : header})
+      .then(function(resp){
+       header.Authorization = 'Bearer ' + resp.body.session_token;
+       chakram.setRequestDefaults({'headers': header});
+     });
+    });
+
+    it('Check webpush service is registered',function(){  
+      var pushService = false;
+
+      return chakram.get(serviceListURL)
+      .then(function(listResponse) {
+        for (var entry of listResponse.body) {
+          if (entry.adapter === 'webpush@link.mozilla.org') {
+              pushService = true;                
+           }
+        }
+        expect(listResponse).to.have.status(200);
+        expect(pushService).equals(true);
+      });
+    });
+
+    it('Add push subscription',function(){     
+      var pushURI = Prepper.webPush_server.getEndpointURI();
+      var pushkey = Prepper.webPush_server.getPublicKey();
+      var setter = 'setter:subscribe.webpush@link.mozilla.org';
+      var payload = {
+        'select': {'id':setter}, 
+        'value': 
+          {'Json': 
+            {'subscriptions':[{
+              'push_uri':pushURI,
+              'public_key':pushkey}]}}};
+
+        return chakram.put(setterURL,payload)
+        .then(function(cmdResp){
+         expect(cmdResp).to.have.status(200);
+         expect(cmdResp.body[setter]).equals(null);
+       });
+    });
+    
+    it('Verify push subscription',function(){   
```

Comment:
I would add more details in the title, 'Add push subscription' also does some verifications